src/common.rs

```diff
@@ -187,147 +43,18 @@ impl FromStr for Header {
-        &self.0
-    }
-
-    pub fn equiv(&self, other: &'static str) -> bool {
-        other.eq_ignore_ascii_case(self.as_str().as_str())
-    }
-}
-
-impl FromStr for HeaderField {
-    type Err = ();
-
-    fn from_str(s: &str) -> Result<HeaderField, ()> {
-        if s.contains(char::is_whitespace) {
-            Err(())
-        } else {
-            AsciiString::from_ascii(s).map(HeaderField).map_err(|_| ())
-        }
-    }
-}
-
-impl Display for HeaderField {
-    fn fmt(&self, formatter: &mut Formatter<'_>) -> Result<(), fmt::Error> {
-        write!(formatter, "{}", self.0.as_str())
-    }
-}
-
-impl PartialEq for HeaderField {
-    fn eq(&self, other: &HeaderField) -> bool {
-        let self_str: &str = self.as_str().as_ref();
-        let other_str = other.as_str().as_ref();
-        self_str.eq_ignore_ascii_case(other_str)
-    }
-}
-
-/// HTTP request methods
-///
-/// As per [RFC 7231](https://tools.ietf.org/html/rfc7231#section-4.1) and
-/// [RFC 5789](https://tools.ietf.org/html/rfc5789)
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub enum Method {
-    /// `GET`
-    Get,
-
-    /// `HEAD`
-    Head,
-
-    /// `POST`
-    Post,
-
-    /// `PUT`
-    Put,
-
-    /// `DELETE`
-    Delete,
-
-    /// `CONNECT`
-    Connect,
-
-    /// `OPTIONS`
-    Options,
-
-    /// `TRACE`
-    Trace,
-
-    /// `PATCH`
-    Patch,
-
-    /// Request methods not standardized by the IETF
-    NonStandard(AsciiString),
-}
-
-impl Method {
-    pub fn as_str(&self) -> &str {
-        match *self {
-            Method::Get => "GET",
-            Method::Head => "HEAD",
-            Method::Post => "POST",
-            Method::Put => "PUT",
-            Method::Delete => "DELETE",
-            Method::Connect => "CONNECT",
-            Method::Options => "OPTIONS",
-            Method::Trace => "TRACE",
-            Method::Patch => "PATCH",
-            Method::NonStandard(ref s) => s.as_str(),
-        }
-    }
-}
-
-impl FromStr for Method {
-    type Err = ();
-
-    fn from_str(s: &str) -> Result<Method, ()> {
-        Ok(match s {
-            "GET" => Method::Get,
-            "HEAD" => Method::Head,
-            "POST" => Method::Post,
-            "PUT" => Method::Put,
-            "DELETE" => Method::Delete,
-            "CONNECT" => Method::Connect,
-            "OPTIONS" => Method::Options,
-            "TRACE" => Method::Trace,
-            "PATCH" => Method::Patch,
-            s => {
-                let ascii_string = AsciiString::from_ascii(s).map_err(|_| ())?;
-                Method::NonStandard(ascii_string)
-            }
-        })
-    }
-}
-
-impl Display for Method {
-    fn fmt(&self, formatter: &mut Formatter<'_>) -> Result<(), fmt::Error> {
-        write!(formatter, "{}", self.as_str())
+    fn fmt(&self, _formatter: &mut Formatter<'_>) -> Result<(), fmt::Error> {
+        // XXX(cosmic): `http` likely intentionally doesn't impl this, so we probably shouldn't
+        // either
+        todo!();
     }
 }
```

Comment:
Seems like we should probably remove this `impl` since `http` allows for arbitrary header field bytes